dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/event/AssertModel.java

```diff
@@ -0,0 +1,13 @@
+package com.dat3m.dartagnan.verification.model.event;
+
+import com.dat3m.dartagnan.program.event.core.Assert;
+import com.dat3m.dartagnan.verification.model.ThreadModel;
+
+import java.util.Set;
+
+
+public class AssertModel extends DefaultEventModel implements RegReaderModel {
+    public AssertModel(Assert event, ThreadModel thread, int id) {
+        super(event, thread, id);
+    }
+}
```

Comment:
Should we store if the assert failed or not? This fact is in principle recomputable on demand but might be a little hassle. Do your witnesses currently mark failed assertions somehow (I think this would be nice for the user)?

Reply:
>Do your witnesses currently mark failed assertions somehow (I think this would be nice for the user)?

Yes. The default `String` of `Assert` indicates the result already. Is it necessary to store it explicitly in `AssertModel`?

Reply:
How does it do so? The default string is in general `assert(expr)` and then the user has to compute `expr` by himself (i.e. figuring out all the register values) to understand if the assertion holds or not.
Don't let yourself get fooled by the fact that the compiler almost always generates `assert(0)`, i.e., `expr = 0`.
Also, the default string can never ever indicate a concrete result unless the result is constant over all executions.

By marking the failed assertion in the witness I mean that you, e.g., mark the assertion node in red, so it is immediately clear where the actual bug is.
That being said, you don't need to do this in this PR. These are just suggestions for the visualization.

Reply:
Independently of the visualization, I agree that the model should be extended to contain the information about the expression evaluating to true or false.

Reply:
I did some tests and found that the assertions which are true don't exist in `Program` object to be extracted (`Asserts` don't but the corresponding `Loads` do). It means all `Asserts` at the stage of constructing the models are failed ones and shown as `assert(False)` in the graph.

Reply:
That's what I mean by "don't let the compiler fool you". If you place a standard assertion `assert(cond)` in your code, the compiler will generate this:
```
if (!cond) {
     assert(0);
     abort();
}
```
That's why every assertion you see is a failed one. You will also only see a single assertion per thread since the code aborts on failure.

However, this is not how our IR works. We do support `assert(cond)` without such transformations and the user can get such assertions by using `__VERIFIER_assert(cond)` rather than the standard `assert(cond)`.
`__VERIFIER_assert(cond)` is directly translates to `assert(cond)` in our internal IR without producing extra control-flow conditions, so you can see successful asserts and even multiple failing asserts in a single thread (because we do not abort on first failure).